### PR TITLE
solution

### PR DESCRIPTION
**Pull request by kostrubitskii**

### CODE REVIEW

**Review by vpashko — APPROVED** [review]
Looks good, don't forget to add a link to the deployed page to make it easier to check the task.

File: src/components/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import React, { useContext } from 'react';
+import classNames from 'classnames';
+import { TodoContext } from '../contexts/TodoContext';
+import { Todo } from '../types/Todo';
+import { FilterStatus } from '../types/Status';
+
+export const Footer: React.FC = () => {
+  const { todos, setStatus, status } = useContext(TodoContext);
+  const visibleTodo = todos.some(todo => todo.completed);
```
Comment (olya-shyian) [inline src/components/Footer.tsx:9]: if we have boolean value its better to start name with `is` or `has`
`hasVisibleTodo`

File: src/components/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React from 'react';
+
+export const Header: React.FC = () => {
+  return (
```
Comment (olya-shyian) [inline src/components/Header.tsx:5]: when you don't have any logic inside components and just return jsx then you can delete the word `return` 

and just return jsx in round brackets

File: src/components/TodoApp.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+export const TodoApp: React.FC = () => {
+  const { todos, error, setError } = useContext(TodoContext);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setError('');
+    }, 3000);
+
+    return () => clearTimeout(timer);
+  }, [error, setError]);
```
Comment (olya-shyian) [inline src/components/TodoApp.tsx:18]: create a component `ErrorNotification` for error jsx and put this logic with setTimeout inside  `ErrorNotification`  component

File: src/components/TodoItem.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className={classNames('todo__status', { completed })}
+          checked={completed}
+        />
+      </label>
+
+      <span data-cy="TodoTitle" className="todo__title">
+        {items.title}
```
Comment (olya-shyian) [inline src/components/TodoItem.tsx:27]: if you used destructuring for `completed` , then do the same for `title`

**Review by olya-shyian — APPROVED** [review]
Well done! 🙂
Please don't forget to add demo link for the next task and fix linter errors

I'll approve. In order to save your time you can do updates according to my comments in the next task

**Pull request closed without merging**